### DISCUSSION TITLE
Wazuh Server as a Primary and Secondary environment

### DISCUSSION THREAD

**Discussion by diveshu** [wazuh/wazuh#16061] [Q&A] [answered]
Can Wazuh Server be implemented as a Primary and Secondary server concept.
For eg:
1) A primary Wazuh server will be the Central server for one or more Wazuh Server which will be considered as Secondary server(s).
2) Agent nodes will be connecting to the Secondary servers and thereby secondary server connects to the Primary server.

**Comment by jctello** [comment, marked as answer]
Hi @diveshu,

Yes, Wazuh has been built with the capability to adapt to the needs of different environments and through the cluster features of both the Wazuh manager and the Wazuh indexer you may configure an architecture to fit your needs.

For example, you may have a Wazuh worker manager receiving communications from all agents and reporting to an Indexer for visualization while a Wazuh master can be used solely for managing the cluster's configuration. This master node may be included in the same instance as the Wazuh Indexer if needed. 

![2managers1indexer](https://user-images.githubusercontent.com/36071202/216001726-827924bb-b339-43f7-a083-37a1b8cc293e.png)


More information on the Wazuh Manager cluster can be found here: https://documentation.wazuh.com/current/user-manual/configuring-cluster/index.html

If you wish to have events indexed to a local Wazuh Indexer and replicated to a secondary Wazuh Indexer this is also possible through Cross Cluster Replication, since the Wazuh Indexer is based on OpenSearch you may see this documentation for reference: https://opensearch.org/docs/1.2/replication-plugin/index/

**Reply by diveshu** [reply]
Hi @jctello, thank you for the information.
